channeldb/channel.go

```diff
@@ -614,6 +625,8 @@ func fetchOpenChannel(chanBucket *bolt.Bucket,
 		return nil, fmt.Errorf("unable to fetch chan revocations: %v", err)
 	}
 
+	channel.Packager = NewChannelPackager(channel.ShortChanID)
```

Comment:
I find it a bit weird that the `OpenChannel` struct holds a `Packager`, that is just a wrapper around the `ShortChanID` (which is already a field). Would it make sense to just make `OpenChannel` adhere to the `FwdPackager` interface?

Reply:
It is weird, I agree 😔

Primary reasons were that some of the methods in the `FwdPackager` iface should only be accessible from within the state change methods. The interface is intended to act as a "persistent mix-in", that requires an appropriate `*bolt.Tx` context. To prevent unintended usage outside of the state changes, I felt it was safer to expose these mixins internally to the channel, and then selectively allow the channel to reveal them at the appropriate granularity, e.g. `LoadFwdPkgs` and `RemoveFwdPkg` where it can provide the proper `*bolt.Tx` context. Would be happier if we found something cleaner :S